EmailNotifications/Email.cs

```diff
@@ -38,72 +39,111 @@ private static void SendCompletedCallback(object sender, AsyncCompletedEventArgs
             mailSent = true;
         }
 
+        public void SendMail(Recipient[] recipient)
+        {
+            if (recipient != null)
+            {
+                string email = ConstructEmail();
+                if (email == "ERROR")
+                {
+                    throw new Exception("Error : No Values Given");
+                }
+                else if (recipient.Length < 1)
+                {
+                    throw new Exception("Error : No Address Given");
+                }
+                else
+                {
+                    try
+                    {
+                        var client = new SmtpClient("smtp.gmail.com", 587)
+                        {
+                            //TODO: Move email address to config file
+                            Credentials = new NetworkCredential("nmotsumi@retrorabbit.co.za", _conf.GetSection("Password").Value),
+                            EnableSsl = true
+                        };
+
+                        //Set From
+                        MailAddress from = new MailAddress(_conf.GetSection("Sender").Value, _conf.GetSection("SenderName").Value);
+
+                        //Set To
+                        MailAddress to = new MailAddress(recipient[0].Address, recipient[0].Name);
+                        MailMessage message = new MailMessage(from, to);
+                        for (int i = 1; i < recipient.Length; i++)
+                        {
+                            message.CC.Add(new MailAddress(recipient[i].Address, recipient[i].Name));
+                        }
+
+                        message.Subject = "WaterLog Notification";
+                        message.IsBodyHtml = true;
+                        message.Body = email;
+                        client.Send(message);
+                    }
+                    catch (Exception e)
+                    {
+                        Console.WriteLine("{0} Exception caught.", e);
```

Comment:
Console logs

Reply:
Threw an exception instead of Console.Writeline.